File: tasks/2025/round2/alienabc.cpp
// @check-accepted: *
// implementation

#include <iostream>
#include <vector>
using namespace std;

string solv() {
    string S;
    cin >> S;
    vector<bool> one(26, true);
    vector<bool> two(26, true);
    S += '*';
    string res = "";
    int i = 0;
    while (i < S.size() - 1) {
        if (S[i] == S[i + 1]) {
            if (two[S[i] - 'a']) {
                two[S[i] - 'a'] = false;
                res += S[i];
                res += S[i + 1];
                res += ' ';
                i += 2;
            } else {
                return "-1";
            }
        } else {
            if (one[S[i] - 'a']) {
                one[S[i] - 'a'] = false;
                res += S[i];
                res += ' ';
                i += 1;
            } else {
                return "-1";
            }
        }
    }
    return res;
}

int main() {
    int T;
    cin >> T;
    for (int i = 0; i < T; i++) {
        cout << solv() << "\n";
    }
    return 0;
}

File: tasks/2025/round2/alienabc.py
#!/usr/bin/env python3
# @check-accepted: *

abc = list(map(chr, range(ord('a'), ord('z')+1)))

def solv():
    string = input()
    if len(string) > 3*26:
        return -1
    for char in abc:
        c = string.count(char)
        if c > 3:
            return -1
        if c > 1 and string.count(2*char) != 1:
            return -1
    res = []
    i = 0
    while i < len(string):
        if i+1 < len(string):
            if string[i]==string[i+1]:
                res.append(2*string[i])
                i += 2
            else:
                res.append(string[i])
                i += 1
        else:
            res.append(string[i])
            i += 1
    return ' '.join(res)

for _ in range(int(input())):
    print(solv())
